tests/test_asyncio.py

```diff
@@ -287,6 +287,11 @@ def test_connect_timeout(self):
                 )
             )
 
+    def test_connect_local_port(self):
```

Comment:
Thank you taking the time to write a test, it makes we want to get this merged :)